tests/difference/core/quint_model/ccv_test.qnt

```diff
@@ -0,0 +1,255 @@
+            GetEmptyProtocolState,
+            "validator",
+            5
+        )
+        val finalResult = votingPowerChange(
+            tmpResult.newState,
+            "validator",
+            0
+        )
+        not(finalResult.hasError) and
+        finalResult.newState.providerState.chainState.currentValidatorSet.get("validator") == 0
+    }
+
+    run VotingPowerSetsFlagTest =
+    {
+        // change voting power
+        val tmpResult = votingPowerChange(
+            GetEmptyProtocolState,
+            "validator",
+            5
+        )
+        // but then set it back to the initial value
+        val finalResult = votingPowerChange(
+            tmpResult.newState,
+            "validator",
+            0
+        )
+        // still should set the flag
+        not(finalResult.hasError) and
+        finalResult.newState.providerState.providerValidatorSetChangedInThisBlock
+    }
+
+
+    // make sure that VotingPowerChange ONLY changes the current validator set, not the history
+    run VotingPowerChangeDoesNotChangeHistoryTest =
+    {
+        val result = votingPowerChange(
+            GetEmptyProtocolState,
+            "validator",
+            0
+        )
+        not(result.hasError) and
+        result.newState.providerState.chainState.votingPowerHistory == List()
+    }
+
+    // Defines a test state to test the deliverPacketToProvider function against.
+    pure val _DeliverPacketToProvider_TestState =
+        val currentState = GetEmptyProtocolState
+        val sender = "sender"
+        val providerState = currentState.providerState
+        val consumerState = GetEmptyConsumerState
+        // add the consumer to the consumerStates
+        val consumerStates = currentState.consumerStates.put(sender, consumerState)
+        val providerState2 = providerState.with(
+            "consumerStatus", providerState.consumerStatus.put(sender, RUNNING)
+        )
+        val providerState3 = providerState2.with(
+            "outstandingPacketsToConsumer", providerState2.outstandingPacketsToConsumer.put(sender, List({
+                id: 0,
+                validatorSet: Map(),
+                sendingTime: 0
+            }))
+        )
+        currentState.with(
+            "providerState", providerState3
+        ).with(
+            "consumerStates", consumerStates
+        )
+
+    // add a packet on the consumer
+    pure val DeliverPacketToProviderHappyPathTest_testState = _DeliverPacketToProvider_TestState.with(
+        "consumerStates", _DeliverPacketToProvider_TestState.consumerStates.put(
+            "sender", _DeliverPacketToProvider_TestState.consumerStates.get("sender").with(
+                "outstandingPacketsToProvider", List({
+                    id: 0,
+                    sendingTime: 0
+                })
+            )
+        )
+    ).with(
+        // put an entry into sentVscPacket on the provider that corresponds to the packet we put on the consumer
```

Comment:
```suggestion
        // put an entry into sentVscPacketsToConsumer on the provider that corresponds to the packet we put on the consumer
```
I guess the packet put on the consumer is the maturation packet?

Reply:
the VSC packet is sent to the consumer, the consumer responds with a VSCMatured packet